common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -76,9 +76,8 @@
     static final InternalLogger logger =
             InternalLoggerFactory.getInstance(HashedWheelTimer.class);
 
-    private static final ResourceLeakDetector<HashedWheelTimer> leakDetector =
-            new ResourceLeakDetector<HashedWheelTimer>(
-                    HashedWheelTimer.class, 1, Runtime.getRuntime().availableProcessors() * 4);
+    private static final ResourceLeakDetector<HashedWheelTimer> leakDetector = ResourceLeakDetectorFactory.instance()
+            .newResourceLeakDetector(HashedWheelTimer.class, 1, Runtime.getRuntime().availableProcessors() * 4);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this multiplication operation to a "long". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2184)